src/ai/backend/manager/scheduler/predicates.py

```diff
@@ -237,3 +243,78 @@ async def check_domain_resource_limit(
             ),
         )
     return PredicateResult(True)
+
+
+async def check_pending_session_limit(
+    db_sess: SASession,
+    sched_ctx: SchedulingContext,
+    sess_ctx: SessionRow,
+) -> PredicateResult:
+    result = True
+    failure_msgs = []
+
+    kernel_query = (
+        sa.select([kernels.c.occupied_slots])
+        .select_from(kernels)
+        .where(
+            (kernels.c.status == KernelStatus.PENDING)
+            & (kernels.c.cluster_role == DEFAULT_ROLE)
+            & (kernels.c.access_key == sess_ctx.access_key)
+        )
+    )
+    pending_kernels: list[dict[str, Any]] = (await db_sess.execute(kernel_query)).fetchall()
+
+    j = sa.join(
+        keypair_resource_policies,
+        keypairs,
+        keypair_resource_policies.c.name == keypairs.c.resource_policy,
+    )
+    policy_query = (
+        sa.select(
+            [
+                keypair_resource_policies.c.max_pernding_sessions,
+                keypair_resource_policies.c.max_pending_session_resource_slots,
+            ]
+        )
+        .select_from(j)
+        .where(keypairs.c.access_key == sess_ctx.access_key)
+    )
+    pending_session_policy: dict[str, Any] = (await db_sess.execute(policy_query)).first()
+
+    max_pending_session: Optional[int] = pending_session_policy["max_pernding_sessions"]
+    if max_pending_session is not None and max_pending_session > 0:
+        if len(pending_kernels) >= max_pending_session:
+            result = False
+            failure_msgs.append(
+                f"You cannot create more than {max_pending_session} pending session."
+            )
+
+    max_pending_session_resource: Optional[ResourceSlot] = pending_session_policy[
+        "max_pernding_sessions"
+    ]
+
+    if max_pending_session_resource:
```

Comment:
What if all resource slot values are 0?
Isn't it different from `null` value?
If so, it should do the explicit `None` check like the count check above.